### ISSUE TITLE
[Question] Explaination needed for Agent Tools

### ISSUE THREAD

**Issue by ShadowArcanist** [Mintplex-Labs/anythingllm-docs#17]
At the moment I am writing the guide for AI Agents and I need some clarifications.... 

![image](https://github.com/Mintplex-Labs/anythingllm-docs/assets/162910371/54398b38-e480-4a20-ae89-3b9bc8202707)

**Summarize Documents**
Agent will give you a summary of the docs and you have to ask it like "Summarize tesla.pdf"

**Web Browsing**
Agent will browser the internet based on the provider we have on agent configuration (Google or Serper) and you have to ask it like **Can you search the internet and tell me about Tesla Cyber Truck**

**Web Scraping**
Agent will scrape the website and tell you the contents on the website and you have to ask it like **Scrape useanything.com and give me the features listed on the website**

**Save File**
Agent will save the file to your local machine and you have to ask it like **Save it as pdf with the name spacex.pdf**

So what does the following tools does and how to make the agent use them?
- RAG Search
- List Documents
- Chart Generation

Also please correct me If I have explained anything wrong about the other tools above (:

**Comment by timothycarambat** [comment]
**RAG Search:**
- Can run a query to check its local memory to answer the user prompt. **requires some data to be embedded in workspace**
eg: `Can you check what you already know about AnythingLLM?`
- Can **update its own memory** and that can be later used for recall in agent **or** regular chat. This basically embeds a virtual document you cannot manage.
eg: `Ah, great point. Can you summarize and save that summary for later to you memory`

List documents can be omitted, its really a helper for doc-summary

**Chat generation**
Can produce a downloaded PNG and UI element in the chat given some data or question
eg: `@agent, can you plot y=mx+b where m=10 and b=0?`
eg: `@agent, can you look at data.csv and plot that as a pie chat by age?`
_assuming data.csv is in the workspace_

**Comment by ShadowArcanist** [comment]
Thanks for the explanation @timothycarambat (:
